### PR TITLE
parser: use identifier as user's resource group name

### PR DESCRIPTION
**Pull request by BornChanger**
Signed-off-by: BornChanger <dawn_catcher@126.com>

### What problem does this PR solve?

Issue Number: ref #38825 

Problem Summary:

### What is changed and how it works?

This pr addressed 2 parser problems related to resource group 
1. Use identifier instead of strLiteral to make user dcl statement easier
2. Fix the drop resource group statement issue

### Check List

Tests 

- [X] Unit test

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by nolouch — APPROVED** [review]

File: parser/parser_test.go [hunk 1]
```diff
@@ -4527,10 +4527,8 @@ func TestPrivilege(t *testing.T) {
 		{`CREATE USER 'root'@'localhost' IDENTIFIED BY 'new-password'`, true, "CREATE USER `root`@`localhost` IDENTIFIED BY 'new-password'"},
 		{`CREATE USER 'root'@'localhost' IDENTIFIED BY PASSWORD 'hashstring'`, true, "CREATE USER `root`@`localhost` IDENTIFIED WITH 'mysql_native_password' AS 'hashstring'"},
 		{`CREATE USER 'root'@'localhost' IDENTIFIED BY 'new-password', 'root'@'127.0.0.1' IDENTIFIED BY PASSWORD 'hashstring'`, true, "CREATE USER `root`@`localhost` IDENTIFIED BY 'new-password', `root`@`127.0.0.1` IDENTIFIED WITH 'mysql_native_password' AS 'hashstring'"},
-		{`CREATE USER 'root'@'127.0.0.1' IDENTIFIED BY 'hashstring' RESOURCE GROUP 'rg1'`, true, "CREATE USER `root`@`127.0.0.1` IDENTIFIED BY 'hashstring' RESOURCE GROUP 'rg1'"},
```
Comment (glorv) [inline parser/parser_test.go:4530]: Why remove this case? You can keep this test as following:
```
{`CREATE USER 'root'@'127.0.0.1' IDENTIFIED BY 'hashstring' RESOURCE GROUP rg1`, true, "CREATE USER `root`@`127.0.0.1` IDENTIFIED BY 'hashstring' RESOURCE GROUP `rg1`"},
```

**Review by glorv — COMMENTED** [review]
rest LGTM

File: parser/parser_test.go (hunk 1 above)
Comment (BornChanger) [reply parser/parser_test.go:4530]: restored sql will add quota marker ` to the resource group name, and the restored sql statement makes parsing fail. And I didn't find a proper way to workaround the parser testing framework at present. So, I moved the syntax check out now.

**Review by tiancaiamao — APPROVED** [review]

File: parser/parser_test.go (hunk 1 above)
Comment (BornChanger) [reply parser/parser_test.go:4530]: Good catch. The problem is resolved now.

**Pull request merged**